src/dvc_objects/fs/base.py

```diff
@@ -136,6 +136,25 @@ def _strip_protocol(cls, path: str) -> str:
     def unstrip_protocol(self, path: str) -> str:
         return path
 
+    def _patch_glob(
+        self,
+        from_info: str,
+        to_info: str,
+        from_infos: List[str],
+    ) -> Tuple[List[str], List[str]]:
+        from .local import localfs
+
+        patched_from_infos = [path.replace("[", "*") for path in from_infos]
+
+        to_infos = [
+            localfs.path.join(
+                to_info, *self.path.relparts(info.replace("*", "["), from_info)
+            )
```

Comment:
It looks like this will break the case where the original path contains a `*` character, since we would end up incorrectly replacing it with the brace.

We either need to properly escape and unescape everything that goes into fsspec's `expand_path` or just dropping (or bypassing) the fsspec glob behavior entirely.

Reply:
I see, good point.